api/audit/views.py

```diff
@@ -64,16 +92,53 @@ def _get_base_filters(self) -> Q:
             project__organisation__userorganisation__role=OrganisationRole.ADMIN,
         )
 
+    def _get_organisation(self) -> Organisation | None:
+        """
+        This is a bit of a hack but since this endpoint is no longer used (by the UI
+        at least) we just return the first organisation the user has (most users only
+        have a single organisation anyway).
+
+        Since this function is only used here for limiting access (by number of days)
+        to the audit log, the blast radius here is pretty small.
+
+        Since we're applying the base filters to the query set
+        """
+        return (
+            self.request.user.organisations.filter(
+                userorganisation__role=OrganisationRole.ADMIN
+            )
+            .select_related("subscription", "subscription_information_cache")
+            .first()
+        )
+
 
 class OrganisationAuditLogViewSet(_BaseAuditLogViewSet):
     permission_classes = [IsAuthenticated, OrganisationAuditLogPermissions]
 
     def _get_base_filters(self) -> Q:
         return Q(project__organisation__id=self.kwargs["organisation_pk"])
 
+    def _get_organisation(self) -> Organisation | None:
+        return (
+            Organisation.objects.select_related(
+                "subscription", "subscription_information_cache"
+            )
+            .filter(pk=self.kwargs["organisation_pk"])
+            .first()
```

Comment:
The reason I did this is that I preferred this code [here](https://github.com/Flagsmith/flagsmith/pull/4433/files#diff-4e1f59ff543830cbdbcd9a52331502f15bef30c1b71fa2b20e666996f25da7c5R68-R70) vs a `try` / `except` but maybe your suggestion is better tbh.

Reply:
Hmm... actually, it's also because [this implementation](https://github.com/Flagsmith/flagsmith/pull/4433/files#diff-4e1f59ff543830cbdbcd9a52331502f15bef30c1b71fa2b20e666996f25da7c5R95-R112) _has_ to be `.first()` so I thought it was better to keep it consistent rather than having to to a try / except _and_ a null check.